src/components/templates/DataTable/TestingTable.tsx

```diff
@@ -0,0 +1,356 @@
+  ...rest
+}: { indeterminate?: boolean } & HTMLProps<HTMLInputElement>) {
+  const ref = useRef<HTMLInputElement>(null!)
+
+  useEffect(() => {
+    if (typeof indeterminate === 'boolean') {
+      ref.current.indeterminate = !rest.checked && indeterminate
+    }
+  }, [ref, indeterminate])
+
+  return (
+    <input
+      type="checkbox"
+      ref={ref}
+      className={className + ' cursor-pointer'}
+      {...rest}
+    />
+  )
+}
+
+const defaultData: Person[] = [
+  {
+    id: 'xxxx1',
+    firstName: 'tanner',
+    lastName: 'linsley',
+    age: 24,
+    visits: 100,
+    status: 'In Relationship',
+    progress: 50,
+    subRows: [
+      {
+        id: 'xxxx12',
+        firstName: 'tanddy',
+        lastName: 'mikker',
+        age: 40,
+        visits: 40,
+        status: 'married',
+        progress: 80,
+        subRows: [
+          {
+            id: 'xxxx123',
+            firstName: 'kalle',
+            lastName: 'talle',
+            age: 41,
+            visits: 40,
+            status: 'Single',
+            progress: 80,
+          },
+        ],
+      },
+      {
+        id: 'xxxx12',
+        firstName: 'joel',
+        lastName: 'dirte',
+        age: 45,
+        visits: 20,
+        status: 'Complicated',
+        progress: 10,
+      },
+      {
+        id: 'xxxx13',
+        firstName: 'tanner1',
+        lastName: 'linsley',
+        age: 24,
+        visits: 100,
+        status: 'In Relationship',
+        progress: 50,
+      },
+    ],
+  },
+  {
+    id: 'xxxx2',
+    firstName: 'tandy',
+    lastName: 'miller',
+    age: 40,
+    visits: 40,
+    status: 'Single',
+    progress: 80,
+  },
+  {
+    id: 'xxxx3',
+    firstName: 'joe',
+    lastName: 'dirte',
+    age: 45,
+    visits: 20,
+    status: 'Complicated',
+    progress: 10,
+  },
+  {
+    id: 'xxxx4',
+    firstName: 'tanner1',
+    lastName: 'linsley',
+    age: 24,
+    visits: 100,
+    status: 'In Relationship',
+    progress: 50,
+  },
+  {
+    id: 'xxxx5',
+    firstName: 'tandy1',
+    lastName: 'miller',
+    age: 40,
+    visits: 40,
+    status: 'Single',
+    progress: 80,
+  },
+  {
+    id: 'xxxx6',
+    firstName: 'joe1',
+    lastName: 'dirte',
+    age: 45,
+    visits: 20,
+    status: 'Complicated',
+    progress: 10,
+  },
+  {
+    id: 'xxxx7',
+    firstName: 'tanner2',
+    lastName: 'linsley',
+    age: 24,
+    visits: 100,
+    status: 'In Relationship',
+    progress: 50,
+  },
+  {
+    id: 'xxxx8',
+    firstName: 'tandy2',
+    lastName: 'miller',
+    age: 40,
+    visits: 40,
+    status: 'Single',
+    progress: 80,
+  },
+  {
+    id: 'xxxx9',
+    firstName: 'joe2',
+    lastName: 'dirte',
+    age: 45,
+    visits: 20,
+    status: 'Complicated',
+    progress: 10,
+  },
+]
+
+const columnHelper = createColumnHelper<Person>()
+
+const columns = [
+  columnHelper.accessor('id', {
+    header: () => 'Id',
+    cell: ({ row, getValue }) => (
+      //Example for link row cell
+      <div
+        style={{
```

Comment:
**[blocker]** (potentially) do we need inline styles for this div or can we use className ?